File: modules/02-sum-types/reject-swift/reject.swift
// DOES NOT TYPE-CHECK: a switch over an enum must be exhaustive. The missing
// `.blue` case is a compile error, the check Go omits.

enum Color {
  case red
  case green
  case blue
}

func name(_ c: Color) -> String {
  switch c {
  case .red: return "red"
  case .green: return "green"
  }
}

File: modules/02-sum-types/reject-kotlin/reject.kt
// DOES NOT COMPILE: a `when` over a sealed type, used as an expression, must be
// exhaustive. The missing `Blue` branch is a compile error.

sealed interface Color

object Red : Color

object Green : Color

object Blue : Color

fun name(c: Color): String =
    when (c) {
        is Red -> "red"
        is Green -> "green"
    }
